lib/teleterm/config.go

```diff
@@ -41,17 +42,21 @@ func (c *Config) CheckAndSetDefaults() error {
 		return trace.BadParameter("missing home directory")
 	}
 
+	if c.CertsDir == "" {
+		return trace.BadParameter("missing certs directory")
+	}
+
 	if c.Addr == "" {
-		c.Addr = fmt.Sprintf("unix://%v/tshd.socket", c.HomeDir)
+		return trace.BadParameter("missing network address")
 	}
 
 	addr, err := utils.ParseAddr(c.Addr)
 	if err != nil {
 		return trace.Wrap(err)
 	}
 
-	if addr.Network() != "unix" {
-		return trace.BadParameter("only unix sockets are supported")
+	if !(addr.Network() == "unix" || addr.Network() == "tcp") {
+		return trace.BadParameter("network address should start with unix:// or tcp:// or be empty (tcp:// is used in that case)")
```

Comment:
The error message seems outdated as in `apiserver.go` we do `lis, err := net.Listen(uri.Network(), uri.Addr)`, no?

Or is this how `uri` is actually going to behave if there's no network specified in the input string?

Reply:
`ParseAddr` function adds `tcp` automatically to the parsed address:
```go
if !strings.Contains(a, "://") {
     a = "tcp://" + a
}
```
so we can pass the address without a network.